### PR TITLE
34 refactor parser

### PR DESCRIPTION
**Pull request by philippschwarzmueller**
Small refactor of the parser using its own header and making the functions static where possible

### CODE REVIEW

File: include/shell.h [hunk 1]
```diff
@@ -46,10 +22,6 @@ void			init_signalhandlers(void);
 t_list			*parse(t_list *lexed_args);
 struct s_state	init_state(void);
```
Comment (oph-design) [inline include/shell.h:23]: Excusez MOI
This has to be in the parser.h file

File: include/shell.h [hunk 2]
```diff
@@ -12,31 +12,7 @@
...
-	char	*path;
-	char	*command;
-	char	**options;
-	int		in;
-	int		out;
-}	t_command;
+# include "parser.h"
 
 t_list			*lexing(char *str);
 void			tokenize_lst(t_list **lst);
```
Comment (oph-design) [inline include/shell.h:18]: Can be removed

**Review by oph-design — CHANGES_REQUESTED** [review]
See my comments

**Comment by philippschwarzmueller** [comment]
fixed the mentioned issues with my latest commit, ty for pointing them out

**Review by oph-design — APPROVED** [review]
LGTM

**Pull request merged**